packages/fxa-auth-server/fxa-oauth-server/lib/routes/token.js

```diff
@@ -105,10 +106,16 @@ const PAYLOAD_SCHEMA = Joi.object({
     .default(MAX_TTL_S)
     .optional(),
 
-  scope: validators.scope.when('grant_type', {
-    is: Joi.string().valid(GRANT_REFRESH_TOKEN, GRANT_FXA_ASSERTION),
-    otherwise: Joi.forbidden(),
-  }),
+  scope: Joi.alternatives()
+    .when('grant_type', {
+      is: GRANT_REFRESH_TOKEN,
+      then: validators.scope.optional(),
+    })
+    .when('grant_type', {
+      is: GRANT_FXA_ASSERTION,
+      then: validators.scope.default(ScopeSet.fromArray([])),
```

Comment:
I think it would be reasonable to make `scope` required when doing an fxa-assertion grant, but didn't want to risk breaking in some edge-case where we're not doing that currently.

Reply:
Similar to the above, I think it should be fine to require scopes for assertion grants since we require a scope of min length 1 on the front end.